osu.Game/Users/UserActivity.cs

```diff
@@ -38,6 +41,11 @@ protected InGame(IBeatmapInfo beatmapInfo, IRulesetInfo ruleset)
             }
 
             public override string Status => Ruleset.CreateInstance().PlayingVerb;
+
+            public override string GetDetails(DiscordRichPresenceMode privacyMode)
+            {
+                return BeatmapInfo.ToString();
```

Comment:
Please do not rely on `.ToString()` to convert an `IBeatmapInfo` to its textual representation. Please use the [`GetDisplayString()`](https://github.com/ppy/osu/blob/ff0c5121e401ba439589becf5a1b325a8e804ff6/osu.Game/Extensions/ModelExtensions.cs#L39) extension method instead.

This goes for every other case of this in this pull (even the implicit ones that do a string conversion via concatenation with the `+` operator).